src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -245,11 +245,23 @@ private string CreateDirectoryIfDoesntExist(string targetPath)
             return fullPath;
         }
 
-        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string pipelineVersionToDownload, string project, string[] tagFilters, string branchName = null)
+        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string pipelineVersionToDownload, string project, string[] tagFilters, string branchName = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var definitions = new List<int>() { Int32.Parse(pipelineDefinition) };
+            if(String.IsNullOrEmpty(pipelineDefinition)) 
+            {
+                throw new InvalidOperationException("Pipeline definition cannot be null or empty");
+            }
+
             VssConnection connection = context.VssConnection;
             BuildHttpClient buildHttpClient = connection.GetClient<BuildHttpClient>();
+
+            var isDefinitionNum = Int32.TryParse(pipelineDefinition, out int definition);
+            if(!isDefinitionNum) 
+            {
+                definition = buildHttpClient.GetDefinitionsAsync(new System.Guid(project), pipelineDefinition, cancellationToken: cancellationToken).SyncResult().FirstOrDefault().Id;
+            }
```

Comment:
This is a little odd, so if we can't extract the definition id from the definition, we ask the build client for it - why not just do that always then? Why two ways of doing the same thing?

Reply:
No sure what you mean. So the issue was that the input from the client is not always the definition ID. Sometime it could just be the def name. For that we have to get the def into from build client. It's the same behavior just like build artifact task.